src/Entity/SectionAssociation.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Drupal\workbench_access\Entity;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+
+/**
+ * Defines the workbench_access SectionAssociation class.
+ *
+ * @ContentEntityType(
+ *   id = "section_association",
+ *   label = @Translation("Section association"),
+ *   bundle_label = @Translation("Access scheme"),
+ *   bundle_entity_type = "access_scheme",
+ *   internal = TRUE,
+ *   handlers = {
+ *     "access" = "Drupal\workbench_access\SectionAssociationAccessControlHandler",
+ *     "storage" = "Drupal\workbench_access\SectionAssociationStorage",
+ *     "views_data" = "\Drupal\workbench_access\ViewsData"
+ *   },
+ *   admin_permission = "assign workbench access",
+ *   base_table = "section_association",
+ *   data_table = "section_association_field_data",
+ *   revision_table = "section_association_revision",
+ *   revision_data_table = "section_association_field_revision_data",
+ *   translatable = FALSE,
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "uuid" = "uuid",
+ *     "revision" = "vid",
+ *     "bundle" = "access_scheme",
+ *   }
+ * )
+ *
+ * @internal
+ *   This entity is marked internal because it should not be used directly.
+ */
+class SectionAssociation extends ContentEntityBase implements SectionAssociationInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
+    $fields = parent::baseFieldDefinitions($entity_type);
+
+    // Assigned users.
+    $fields['user_id'] = BaseFieldDefinition::create('entity_reference')
```

Comment:
This should be called users because it is not an ID anymore. It is an entity. Also `->setCardinality` unlimited is still missing.